sympy/solvers/solvers.py

```diff
@@ -2653,22 +2655,50 @@ def _tsolve(eq, sym, **flags):
                     return _solve(lhs - rhs, sym, **flags)
             elif sym not in lhs.exp.free_symbols:
                 return _solve(lhs.base - rhs**(1/lhs.exp), sym, **flags)
-            elif not rhs and sym in lhs.exp.free_symbols:
-                # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
-                # the same place
-                sol_base = _solve(lhs.base, sym, **flags)
-                if not sol_base:
-                    return sol_base  # no solutions to remove so return now
-                return list(ordered(set(sol_base) - set(
-                    _solve(lhs.exp, sym, **flags))))
-            elif (rhs is not S.Zero and
-                        lhs.base.is_positive and
-                        lhs.exp.is_real):
-                return _solve(lhs.exp*log(lhs.base) - log(rhs), sym, **flags)
-            elif lhs.base == 0 and rhs == 1:
-                return _solve(lhs.exp, sym, **flags)
             else:
-                raise NotImplementedError
+                # sym in lhs.exp.free_symbols
+                if not rhs:
+                    # f(x)**g(x) only has solutions where f(x) == 0 and g(x) != 0 at
+                    # the same place
+                    sol_base = _solve(lhs.base, sym, **flags)
+                    if not sol_base:
+                        return sol_base  # no solutions to remove so return now
+                    return list(ordered(set(sol_base) - set(
+                    _solve(lhs.exp, sym, **flags))))
+                elif rhs.is_Rational or lhs.exp.is_real:
+                    sol=[]
+                    if  lhs.base == 0 and rhs==0:
```

Comment:
already handled above with "if not rhs", isn't it?

Reply:
I had taken out this case ```if  lhs.base == 0 and rhs==0:``` because i think we should handle this case differently as above code which is already implemented